jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/jsonb/JsonbTransientAnnotationQuickFix.java

```diff
@@ -22,7 +36,36 @@
  *
  */
 public class JsonbTransientAnnotationQuickFix extends RemoveAnnotationConflictQuickFix {
+
     public JsonbTransientAnnotationQuickFix() {
-        super("jakarta.json.bind.annotation.JsonbTransient");
+        // annotation list to be derived from the diagnostic passed to
+        // `getCodeActions()`
+        super();
+    }
+
+    @Override
+    public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+            IProgressMonitor monitor) throws CoreException {
+        ASTNode node = context.getCoveredNode();
+        IBinding parentType = getBinding(node);
+
+        JsonArray diagnosticData = (JsonArray) diagnostic.getData();
+
+        List<String> annotations = IntStream.range(0, diagnosticData.size())
+                .mapToObj(idx -> diagnosticData.get(idx).getAsString()).collect(Collectors.toList());
+
+        if (parentType != null) {
+            List<CodeAction> codeActions = new ArrayList<>();
+            // We either keep, JsobTransient, or remove all other Jsonb annotations.
+            removeAnnotation(diagnostic, context, parentType, codeActions,
+                    "jakarta.json.bind.annotation.JsonbTransient");
+
+            annotations.remove(JsonbConstants.JSONB_TRANSIENT);
+            removeAnnotation(diagnostic, context, parentType, codeActions,
+                    annotations.toArray(new String[] {}));       
+
+            return codeActions;
```

Comment:
Since much of this logic looks similar to https://github.com/eclipse/lsp4jakarta/blob/566472aba6d6682c9c4272131fff3700c1dd8df4/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/cdi/ScopeDeclarationQuickFix.java#L34-L53
consider moving this logic to a common utility method

Reply:
Where do you suggest placing this common logic, as any method that returns the needed objects must return both `parentType' and `annotations`.
Should I use the template method pattern and crate an abstract class called `RemoveMultipleAnnotationsQucikFix`?

Reply:
Yes I like the latter suggestion of the `RemoveMultipleAnnotations` class, consider placing this in the code actions proposal package: https://github.com/eclipse/lsp4jakarta/tree/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/codeAction/proposal

Reply:
Done